ibm-eventstore/src/test/java/akka/stream/alpakka/ibm/eventstore/javadsl/EventStoreSpec.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.ibm.eventstore.javadsl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.japi.Pair;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.testkit.JavaTestKit;
+import com.ibm.event.catalog.TableSchema;
+import com.ibm.event.common.ConfigurationReader;
+import com.ibm.event.oltp.EventContext;
+import com.ibm.event.oltp.InsertResult;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.junit.*;
+
+
+import static org.junit.Assert.assertEquals;
+
+
+/**
+ * This integration test can be run using a local installation of EventStore
+ * The installer for EventStore can be obtained from:
+ * https://www.ibm.com/us-en/marketplace/project-eventstore
+ *
+ * Note: Run each integration test (Java and Scala) one at the time
+ *
+ * Before running the test:
+ * Change the host and port below in the function 'setEndpoint' to the EventStore
+ * Change the host and port below in the function 'failureEndpoint' to a unresponsive host/port.
+ *
+ */
+@Ignore
+public class EventStoreSpec {
+    private static ActorSystem system;
+    private static Materializer materializer;
+    private static EventContext eventContext;
+    private static String databaseName = "TESTDB";
+    private static String tableName = "TESTTABLE";
+    private static void setEndpoint() {
+        // #configure-endpoint
+        ConfigurationReader.setConnectionEndpoints("127.0.0.1:5555");
```

Comment:
wat? So there is a global mutable context that the IBM thing uses? A single JVM can only connect to one of these at a time? No API for actually representing that context as an object and pass it around instead? I think that if that is possible, and it needs to be shared among stages, it would be much preferred to pass it around to the stream factories instead. If it doesn't require sharing and can be non-jvm-global then maybe even let the stage itself take the connection config and manage the lifecycle?

Reply:
To my knowledge this is the case, a global single endpoint.

https://www.ibm.com/support/knowledgecenter/SSGNPV/eventstore/desktop/dev-guide.html#insertData